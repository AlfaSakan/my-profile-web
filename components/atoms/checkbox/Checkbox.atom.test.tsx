import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox.atom';

describe('buttom component test', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Checkbox
          onChange={() => {}}
          data={{ checked: true, label: 'Label', value: 'label' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Checkbox
          onChange={() => {}}
          data={{ checked: false, label: 'Label', value: 'label' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should onclick called', () => {
    const onClick = jest.fn();
    render(
      <Checkbox
        onChange={onClick}
        data={{ checked: false, label: 'Label', value: 'label' }}
      />
    );
    fireEvent.click(screen.getByText('Label'));
    expect(onClick).toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
