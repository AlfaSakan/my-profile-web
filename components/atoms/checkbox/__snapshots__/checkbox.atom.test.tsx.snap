// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttom component test should render correctly 1`] = `
<div
  className="flex items-center gap-2 cursor-pointer"
  onClick={[Function]}
>
  <input
    checked={true}
    className="cursor-pointer"
    id="label"
    name="label"
    readOnly={true}
    type="checkbox"
    value="label"
  />
  <label
    className="cursor-pointer"
    id="label"
  >
    Label
  </label>
</div>
`;

exports[`buttom component test should render correctly 2`] = `
<div
  className="flex items-center gap-2 cursor-pointer"
  onClick={[Function]}
>
  <input
    checked={false}
    className="cursor-pointer"
    id="label"
    name="label"
    readOnly={true}
    type="checkbox"
    value="label"
  />
  <label
    className="cursor-pointer"
    id="label"
  >
    Label
  </label>
</div>
`;
